packages/form-js-viewer/src/render/components/form-fields/Taglist.js

```diff
@@ -0,0 +1,158 @@
+import { useContext, useEffect, useRef, useState } from 'preact/hooks';
+
+import { FormContext } from '../../context';
+
+import CloseIcon from './icons/Close.svg';
+
+import Description from '../Description';
+import Errors from '../Errors';
+import Label from '../Label';
+
+import {
+  formFieldClasses,
+  prefixId
+} from '../Util';
+import DropdownList from './parts/DropdownList';
+
+const type = 'taglist';
+
+
+export default function Taglist(props) {
+  const {
+    disabled,
+    errors = [],
+    field,
+    value : values = []
+  } = props;
+
+  const {
+    description,
+    id,
+    label,
+    values : options
+  } = field;
+
+  const { formId } = useContext(FormContext);
+  const [filter, setFilter] = useState('');
+  const [selectedValues, setSelectedValues] = useState([]);
+  const [filteredValues, setFilteredValues] = useState([]);
+  const [isDropdownExpanded, setIsDropdownExpanded] = useState(false);
+  const [hasValuesLeft, setHasValuesLeft] = useState(true);
+  const [escapeClose, setEscapeClose] = useState(false);
+  const searchbar = useRef();
+
+
+  // Usage of stringify is necessary here because we want this effect to only trigger when there is a value change to the array
+  useEffect(() => {
+    const selectedValues = values.map(v => options.find(o => o.value === v)).filter(v => v !== undefined);
+    setSelectedValues(selectedValues);
+  }, [JSON.stringify(values), options]);
+
+  useEffect(() => {
+    setFilteredValues(options.filter((o) => o.label && (o.label.toLowerCase().includes(filter.toLowerCase())) && !values.includes(o.value)));
+  }, [filter, JSON.stringify(values), options]);
+
+  useEffect(() => {
+    setHasValuesLeft(selectedValues.length < options.length);
+  }, [selectedValues.length, options.length]);
+
+  const onFilterChange = ({ target }) => {
+    setEscapeClose(false);
+    setFilter(target.value);
+  };
+
+  const selectValue = (option) => {
+    setFilter('');
+    props.onChange({ value: [...values, option.value], field });
+  };
+
+  const deselectValue = (option) => {
+    props.onChange({ value: values.filter((v) => v != option.value), field });
+  };
+
+  const onInputKeyDown = (e) => {
+
+    switch (e.key) {
+    case 'ArrowUp':
+    case 'ArrowDown':
+
+      // We do not want the cursor to seek in the search field when we press up and down
+      e.preventDefault();
+      break;
+    case 'Backspace':
+      if (!filter && selectedValues.length) {
+        deselectValue(selectedValues[selectedValues.length - 1]);
+      }
+      break;
+    case 'Escape':
+      setEscapeClose(true);
+      break;
+    case 'Enter':
+      if (escapeClose) {
+        setEscapeClose(false);
+      }
+      break;
+    }
+  };
+
+  return <div class={ formFieldClasses(type, errors) }>
+    <Label
+      label={ label }
+      id={ prefixId(id, formId) } />
+    <div class={ 'fjs-taglist' + (disabled ? ' disabled' : '') }>
```

Comment:
Could we use `classnames` here? I think a good practice would be to structure the classes like
```js
class={ classnames('fjs-taglist', { 'fjs-taglist-disabled': disabled }) }
```